### ISSUE TITLE
[Accessibility Feature] current mode (plan / act) cannot be recognized by screen readers

### ISSUE THREAD

**Issue by yncat** [cline/cline#4244]
### What happened?

When trying to change between plan and act, screen readers cannot tell which is currently active.

### Steps to reproduce

1. Try to focus on either of plan / act to see which is currently selected
2. Screen readers do not announce the active selection

### Relevant API REQUEST output

```shell

```

### Provider/Model

LiteLLM

### Operating System

Windows

### System Info

Screen readers: NVDA and Narrator tested

### Cline Version

v3.17.13

### Additional context

I'm almost sure that just adding aria-pressed and role="button" solves the issue. So I'm willing to submit a PR if accepted.

**Comment by kevinneung** [comment]
Hi @yncat , please feel free to submit a PR for this. Something done in the style of #4246 , would be great. Thank you!
